File: ABC/abc010/abc010_3.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int tx_a, ty_a, tx_b, ty_b, T, V, n, x, y;
    cin >> tx_a >> ty_a >> tx_b >> ty_b >> T >> V >> n;

    string ans = "NO";
    for (int i = 0; i < n; i++) {
        cin >> x >> y;
        if (sqrt((x - tx_a) * (x - tx_a) + (y - ty_a) * (y - ty_a)) + sqrt((x - tx_b) * (x - tx_b) + (y - ty_b) * (y - ty_b)) <= V * T) ans = "YES";
    }
    cout << ans << endl;
}

File: ABC/abc010/abc010_3.py
tx_a, ty_a, tx_b, ty_b, T, V = map(int, input().split())

n = int(input())
for i in range(n):
    x, y = map(int, input().split())
    if ((x - tx_a) ** 2 + (y - ty_a) ** 2) ** 0.5 + ((x - tx_b) ** 2 + (y - ty_b) ** 2) ** 0.5 <= V * T:
        print("YES")
        break
else:
    print("NO")
